File: Chapter02_GettingStarted/SelectionSort/Python/main.py
import sys


def selection_sort(input):
    for i in range(len(input) - 1):
        min_index = i
        for j in range(i + 1, len(input)):
            if input[j] < input[min_index]:
                min_index = j
        aux = input[i]
        input[i] = input[min_index]
        input[min_index] = aux
    return input


if __name__ == "__main__":
    input = [int(x) for x in sys.argv[1:]]
    selection_sort(input)
    print(input)

File: Chapter02_GettingStarted/SelectionSort/C/main.c
#include<stdio.h>

int* selectionSort(int *input, int length)
{
    for (int i = 0; i < length - 1; i++)
    {
        int j = i + 1;
        int minIndex = j - 1;
        while (j < length)
        {
            if (*(input + j) < *(input + minIndex)) 
            {
                minIndex = j;
            }
            j++;
        }
        int aux = *(input + i);
        *(input + i) = *(input + minIndex);
        *(input + minIndex) = aux;
    }
    return input;
}

void printArray(int* input, int length)
{
    printf("[");
    for (int i = 0; i < length - 1; i++)
    {
        printf("%d, ", *(input + i));
    }
    if (length > 0)
        printf("%d]\n", *(input + length - 1));
    else
        printf("]\n");
}

int main(int argv, char* argc[]) 
{
    int worstCase[] = { 4, 3, 2, 1 };

    selectionSort(worstCase, 4);
    printArray(worstCase, 4);

    int bestCase[] = { 1, 2, 3, 4 };

    selectionSort(bestCase, 4);
    printArray(bestCase, 4);

    return 0;
}